### ISSUE TITLE
RemotaApp   xf_create_window: Assertion `!xfc->gc' failed

### ISSUE THREAD

**Issue by Phafid07** [FreeRDP/FreeRDP#2410]
I have been trying to use remote app from my debian 7 64bit, i compiled the code from the source and i have tryed different sintax, the las one was 
This is FreeRDP version 1.2.1-dev (git ebdcf04)

xfreerdp /u:miusuario /d:INTRANET /p:password  /app:"||winword" /v:M-TS01
That remote app has been used from windows clients and is working

This is the output

user@pn25:/etc$ xfreerdp /u:user /d:INTRANET /p:pass$  /app:"||winword" /v:M-TS01
[10:18:16:132] [23623:23624] [INFO][com.freerdp.client.common.cmdline] - loading channel rail
xfreerdp: /home/pablo/src/FreeRDP/client/X11/xf_client.c:596: xf_create_window: Assertion `!xfc->gc' failed.
Aborted

if i dont use the /app parameter , it works and gets me into a remote desktop

**Comment by geiseri** [comment]
I also see this with master (FreeRDP@c3bbe706fbf0f7c03d409190a483f8429c1e1b10).  I am on Arch 64bit and connecting to Windows 7 Ultimate (32bit).  I ran ```xfreerdp /v:deneb /u:geiseri /d:geekcentral /p:*** /app:"||notepad"```.  The application is properly published and I can connect with mstsc as well as freerdp from the freerdp-1.2.0_beta1+android9-1 tag will connect. Again without the ```/app``` I can connect without any problems.

**Comment by geiseri** [comment]
As an update, I noticed something strange.  It seems to run sometimes, but after a few runs it will fail again. If I connect to the remote app with mstsc and the RDP file, then it works again for a few runs.  Is there a chance that sometimes it won't clean up properly?

**Comment by abaudin** [comment]
Hi,
I've got exactly the same problem. When i try to run my explorer program, in command line (\app:"explorer") and with the .rdp file it failed, same error (but line 600 : xf_client.c:600).
Maybe an option didn't work (like width, colors, souds...) to create the window ?

**Comment by elxa** [comment]
##### GDB backtrace:
```
Reading symbols from ./client/X11/xfreerdp...done.
Starting program: /home/dice/Projects/FreeRDP/client/X11/xfreerdp /cert-ignore /v:172.30.42.15 /u:RemoteApp /p:test /app:"||paint"
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
[New Thread 0x7fffeb5dd700 (LWP 23529)]
xfreerdp: /home/dice/Projects/FreeRDP/client/X11/xf_client.c:600: xf_create_window: Assertion `!xfc->gc' failed.
[17:52:16:396] [23421:23529] [INFO][com.freerdp.client.common.cmdline] - loading channel rail
[New Thread 0x7fffea97c700 (LWP 23549)]

Program received signal SIGABRT, Aborted.
[Switching to Thread 0x7fffeb5dd700 (LWP 23529)]
0x00007ffff293ad6b in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:55
55	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
#0  0x00007ffff293ad6b in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:55
#1  0x00007ffff293c231 in __GI_abort () at abort.c:89
#2  0x00007ffff2933989 in __assert_fail_base (fmt=0x7ffff2a73cb8 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", assertion=assertion@entry=0x7ffff7b56426 "!xfc->gc", file=file@entry=0x7ffff7b563d8 "/home/dice/Projects/FreeRDP/client/X11/xf_client.c", line=line@entry=600, 
    function=function@entry=0x7ffff7b56b50 <__PRETTY_FUNCTION__.19852> "xf_create_window") at assert.c:92
#3  0x00007ffff2933a33 in __GI___assert_fail (assertion=0x7ffff7b56426 "!xfc->gc", file=0x7ffff7b563d8 "/home/dice/Projects/FreeRDP/client/X11/xf_client.c", line=600, function=0x7ffff7b56b50 <__PRETTY_FUNCTION__.19852> "xf_create_window") at assert.c:101
#4  0x00007ffff7b51186 in xf_create_window (xfc=0x62a7a0) at /home/dice/Projects/FreeRDP/client/X11/xf_client.c:600
#5  0x00007ffff7b43858 in xf_rail_disable_remoteapp_mode (xfc=0x62a7a0) at /home/dice/Projects/FreeRDP/client/X11/xf_rail.c:78
#6  0x00007ffff7b536b2 in xf_logon_error_info (instance=0x62a4c0, data=4294967291, type=2) at /home/dice/Projects/FreeRDP/client/X11/xf_client.c:1256
#7  0x00007ffff4a144a4 in rdp_recv_logon_error_info (rdp=0x62ff90, s=0x63dd60) at /home/dice/Projects/FreeRDP/libfreerdp/core/info.c:728
#8  0x00007ffff4a14896 in rdp_recv_logon_info_extended (rdp=0x62ff90, s=0x63dd60) at /home/dice/Projects/FreeRDP/libfreerdp/core/info.c:770
#9  0x00007ffff4a14a9d in rdp_recv_save_session_info (rdp=0x62ff90, s=0x63dd60) at /home/dice/Projects/FreeRDP/libfreerdp/core/info.c:807
#10 0x00007ffff4a5effa in rdp_recv_data_pdu (rdp=0x62ff90, s=0x63dd60) at /home/dice/Projects/FreeRDP/libfreerdp/core/rdp.c:818
#11 0x00007ffff4a5fc3f in rdp_recv_tpkt_pdu (rdp=0x62ff90, s=0x63dd60) at /home/dice/Projects/FreeRDP/libfreerdp/core/rdp.c:1096
#12 0x00007ffff4a60133 in rdp_recv_pdu (rdp=0x62ff90, s=0x63dd60) at /home/dice/Projects/FreeRDP/libfreerdp/core/rdp.c:1173
#13 0x00007ffff4a60766 in rdp_recv_callback (transport=0x63d9d0, s=0x63dd60, extra=0x62ff90) at /home/dice/Projects/FreeRDP/libfreerdp/core/rdp.c:1276
#14 0x00007ffff4a69ec6 in transport_check_fds (transport=0x63d9d0) at /home/dice/Projects/FreeRDP/libfreerdp/core/transport.c:752
#15 0x00007ffff4a609b6 in rdp_check_fds (rdp=0x62ff90) at /home/dice/Projects/FreeRDP/libfreerdp/core/rdp.c:1328
#16 0x00007ffff4a3694b in freerdp_check_fds (instance=0x62a4c0) at /home/dice/Projects/FreeRDP/libfreerdp/core/freerdp.c:208
#17 0x00007ffff4a36a63 in freerdp_check_event_handles (context=0x62a7a0) at /home/dice/Projects/FreeRDP/libfreerdp/core/freerdp.c:242
#18 0x00007ffff7b53e2e in xf_client_thread (param=0x62a4c0) at /home/dice/Projects/FreeRDP/client/X11/xf_client.c:1453
#19 0x00007ffff41683f9 in thread_launcher (arg=0x671780) at /home/dice/Projects/FreeRDP/winpr/libwinpr/thread/thread.c:297
#20 0x00007ffff3104340 in start_thread (arg=0x7fffeb5dd700) at pthread_create.c:310
#21 0x00007ffff29f65ed in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
```

**Comment by elxa** [comment]
##### output of WLOG_LEVEL=DEBUG

https://gist.github.com/anonymous/e7f590a7f487cec9f310+

This still happens with master ( 121ea2359550388b34bc36047eb404f90e504737 ) and works without issues with (x)freerdp 1.0.2 .

**Comment by igyo** [comment]
I have the same problem ( This is FreeRDP version 1.2.1-dev (git 121ea23) )

**Comment by akallabeth** [comment]
Fixed with #4322
